Update instructions since pip has removed the --process-dependency-links option

duncan-brown opened the pull request:
Add the ``--process-dependency-links`` option to pip to bring in the dependencies that PyCBC needs.

samantha-usman commented:
I don't know what dependency links are. What should this option take in? What would the error message be without it?

duncan-brown commented:
@samantha-usman ok, this is more complicated that I thought. ``--process-dependency-links`` was an option to pip that would allow a package from github to bring in another package. This is used to install mpld3 as part of the PyCBC install, see:

https://github.com/ligo-cbc/pycbc/blob/master/setup.py#L73

However, it appears that the pip maintainers have removed this feature in pip 9.0 (see https://github.com/pypa/pip/pull/1519 and https://github.com/pypa/pip/issues/3939).

The solution to this is to add a line to the instructions telling the user to install mpld3, like we do for the segment database tools.

I will git push -f this update with that patch in a few minutes.

duncan-brown commented:
@samantha-usman I have updated the instructions to tell you do a an explicit install of mpld3:
```
pip install https://github.com/ligo-cbc/mpld3/tarball/master#egg=mpld3-0.3git
```
as part of the installation of required packages. I removed the deprecated ``--process-dependency-links`` argument.

@ahnitz I also removed it from the ``setup.py`` file

@samantha-usman please try this and see if it works. Please comment here if it works or fails, or if you are having issues with the instructions.

ahnitz commented on setup.py:
```diff
@@ -70,7 +70,6 @@
                       'corner>=2.0.1',
                       #'scikit-learn>=0.17.0',  # travis does not like scikit-learn
                       ]
```
do not remove from the setup.py. It *is* required afterall. Even if you are installing from outside pycbc, we should not remove the dependency.

duncan-brown replied:
But that directive has been deprecated in both setuptools and pip. We shouldn't have it there if it's a broken noop. If you want it in setup.py, the right thing to do is release a ``pycbc-mpld3`` in PyPi and then add it to the regular requirements.

ahnitz commented:
@duncan-brown I think this is the better solution. 

https://github.com/ligo-cbc/pycbc/pull/1204

I

duncan-brown commented:
I incorporared https://github.com/ligo-cbc/pycbc/pull/1204 here so that it also gets the other fixes to the documentation.
